smartpower3/setting.cpp

```diff
@@ -84,32 +84,42 @@ void Setting::popUp(void)
 	popup->pushSprite(200, 200);
 }
 
-uint8_t Setting::_setBacklightLevel(uint8_t level)
+uint8_t Setting::_setBacklightLevelPreset(uint8_t level_preset)
 {
-	if (level > 6)
-		level = 6;
-	else if (level < 0)
-		level = 0;
-	ledcWrite(2, bl_value[level]);
-	return level;
+	if (level_preset > 6)
+		level_preset = 6;
+	else if (level_preset < 0)
+		level_preset = 0;
+	this->setBacklightLevel(bl_value_preset[level_preset]);
+	return level_preset;
 }
 
-uint8_t Setting::setBacklightLevel(void)
+uint8_t Setting::setBacklightLevelPreset(void)
 {
-	backlight_level = backlight_level_edit;
-	return _setBacklightLevel(backlight_level);
+	backlight_level_preset = backlight_level_edit;
+	return _setBacklightLevelPreset(backlight_level_preset);
 }
 
-void Setting::setBacklightLevel(uint8_t level, bool edit)
+void Setting::setBacklightLevelPreset(uint8_t level_preset, bool edit)
 {
-	backlight_level = _setBacklightLevel(level);
+	backlight_level_preset = _setBacklightLevelPreset(level_preset);
 	if (edit)
-		backlight_level_edit = backlight_level;
+		backlight_level_edit = backlight_level_preset;
 }
 
 void Setting::setBacklightLevel(uint8_t level)
 {
-	backlight_level = _setBacklightLevel(level);
+	if (level > 255) {
+		level = 255;
+	} else if (level < 0) {
+		level = 0;
+	}
+	ledcWrite(2, level);
+}
+
```

Comment:
Purpose of the method has changed from setting preset level (array member value) to directly setting arbitrary value between 0 and 255. Made it a separate method since it is reused (in `turnOffBacklight` and in `_setBacklightLevelPreset`).